src/main/java/org/jenkinsci/plugins/workflow/multibranch/BranchJobProperty.java

```diff
@@ -66,11 +67,17 @@ synchronized void setBranch(@Nonnull Branch branch) {
             @Override public boolean hasPermission(Authentication a, Permission permission) {
                 // This project is managed by its parent and may not be directly configured or deleted.
                 // Note that Item.EXTENDED_READ may still be granted, so you can still see Snippet Generator, etc.
-                if (ACL.SYSTEM.equals(a)) {
+                if (branch instanceof Branch.Dead && (permission == Job.BUILD)) {
+                    // if dead, nobody, not even SYSTEM may build this job
+                    return false;
+                } else if (ACL.SYSTEM.equals(a)) {
                     return true; // e.g., DefaultDeadBranchStrategy.runDeadBranchCleanup
```

Comment:
Is this not now covered by the check below? I guess it is safest to unconditionally allow `SYSTEM` to do what it needs, but would be advisable to update comments to reflect the new reality.